src/ecpool_worker.erl

```diff
@@ -74,17 +74,17 @@ start_link(Pool, Id, Mod, Opts) ->
 client(Pid) ->
     gen_server:call(Pid, client, infinity).
 
--spec(exec(pid(), action()) -> Result :: any() | {error, Reason :: term()}).
-exec(Pid, Action) ->
-    gen_server:call(Pid, {exec, Action}, infinity).
+-spec(exec(pid(), action(), timeout()) -> Result :: any() | {error, Reason :: term()}).
+exec(Pid, Action, Timeout) ->
+    gen_server:call(Pid, {exec, Action}, Timeout).
 
 -spec(exec_async(pid(), action()) -> Result :: any() | {error, Reason :: term()}).
```

Comment:
bad spec, gen_server:cast always returns ok

Reply:
I'll fix this spec